component/metrics/receiver.go

```diff
@@ -0,0 +1,19 @@
+package metrics
+
+import (
+	"github.com/prometheus/prometheus/model/labels"
+	"github.com/prometheus/prometheus/storage"
+)
+
+// Receiver is used to pass an array of metrics to another receiver
+type Receiver struct {
+	// metrics should be considered immutable
+	Receive func(timestamp int64, metrics []FlowMetric) `hcl:"receiver"`
+}
+
+// FlowMetric is a wrapper around a single metric without the timestamp
+type FlowMetric struct {
+	Ref    storage.SeriesRef
```

Comment:
Maybe add the reference id here so on the return you can get the results. The results then need to be synchronous which might be fine?